lib/ble/ble.toit

```diff
@@ -713,6 +727,69 @@ class DataBlock:
     result.replace 2 data
     return result
 
+  hash-code -> int:
+    return type.hash-code * 31 + data.hash-code
+
+  operator== other/any:
+    if other is not DataBlock: return false
+    other-block := other as DataBlock
+    return type == other-block.type and data == other-block.data
+
+  stringify --add-prefix/bool=true -> string:
+    prefix := add-prefix ? "DataBlock: " : ""
+    if type == TYPE-RAW: return "$(prefix)raw=$data"
+    if type == TYPE-FLAGS:
+      flags-string := ""
+      if (flags & BLE-ADVERTISE-FLAGS-LIMITED-DISCOVERY) != 0:
+        flags-string += "LE Limited Discoverable Mode, "
+      if (flags & BLE-ADVERTISE-FLAGS-GENERAL-DISCOVERY) != 0:
+        flags-string += "LE General Discoverable Mode, "
+      if (flags & BLE-ADVERTISE-FLAGS-BREDR-UNSUPPORTED) != 0:
+        flags-string += "BR/EDR Not Supported, "
+      if (flags & BLE-ADVERTISE-FLAGS-LE-BREDR-CONTROLLER) != 0:
+        flags-string += "LE and BR/EDR Controller, "
+      if (flags & BLE-ADVERTISE-FLAGS-LE-BREDR-HOST) != 0:
+        flags-string += "LE and BR/EDR Host, "
+      if flags-string.is-empty: return "$(prefix)flags=$(%x flags)"
+      flags-string = flags-string[.. flags-string.size - 2]
```

Comment:
Are you dropping the last ", " like this? Hmm. At least it deserves a comment, but using join would be a lot nicer.

Reply:
done.